### PR TITLE
DTSPO-17278 further improve shuttering

### PR DESCRIPTION
**Pull request by ieuanb74**
### Jira link (if applicable)
https://tools.hmcts.net/jira/browse/DTSPO-17278

### Change description ###
- Removed zone level tagging and associated logic to reduce complexity
- Applied tagging to individual records
- Renamed area to platform 
- Removed `zone_area` var and references to it 

### Checklist

- [x] commit messages are meaningful and follow good commit message guidelines
- [x] README and other documentation has been updated / added (if needed)
- [x] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

**Comment by endakelly** [comment]
Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?

**Comment by ieuanb74** [comment]
> Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?

Yep good shout. Have also added this to HMCTS way under the section about configuring public dns

**Comment by endakelly** [comment]
> > Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?
> 
> Yep good shout. Have also added this to HMCTS way under the section about configuring public dns
> 
> Actually might not be required as already link to HMCTS way and the example in this repo for more info on tagging - [hmcts/azure-public-dns#shuttering](https://github.com/hmcts/azure-public-dns?tab=readme-ov-file#shuttering)

Yeah, I think the overall guidance is fine, I just don't see an explicit example showing `platform` or `area` being set. Maybe just add a link to a record for cft and sds where the value has been set so people know what it should look like.

**Comment by ieuanb74** [comment]
> > > Do we need some info in the main readme to say that teams should ensure platform is defined or will it assume a default value?
> > 
> > 
> > Yep good shout. Have also added this to HMCTS way under the section about configuring public dns
> > Actually might not be required as already link to HMCTS way and the example in this repo for more info on tagging - [hmcts/azure-public-dns#shuttering](https://github.com/hmcts/azure-public-dns?tab=readme-ov-file#shuttering)
> 
> Yeah, I think the overall guidance is fine, I just don't see an explicit example showing `platform` or `area` being set. Maybe just add a link to a record for cft and sds where the value has been set so people know what it should look like.

I've updated this with a link to example

File: modules/azure-public-dns/a_record.tf [hunk 1]
```diff
@@ -1,18 +1,18 @@
 locals {
   # Extract shutter_all_a value if available or false
   # If global shutter is true, set to true otherwise set to value of shutter_all_a
-  shutter_all_a = local.shutter_all == true ? true : (var.zone_area == "cft" && local.shutter_all_cft == true) ? true : (var.zone_area == "sds" && local.shutter_all_sds == true) ? true : lookup(yamldecode(var.shutter_config), "shutter_all_a", false)
+  shutter_all_a = local.shutter_all == true ? true : lookup(yamldecode(var.shutter_config), "shutter_all_a", false)
```
[the remark this answers is not in the document]
Comment (ieuanb74) [reply modules/azure-public-dns/a_record.tf:4]: No longer required as have removed this in favour of tagging at the record level only. Same experience for user and makes the logic easier to read

**Review by endakelly — APPROVED** [review]

**Pull request merged**